### DISCUSSION TITLE
[help]  Dependency analysis misses some objects

### DISCUSSION THREAD

**Discussion by llrs** [ropensci/targets#1089] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

My targets project has a hard time recognizing some targets:

``` r
library("targets")
packageVersion("targets")
#> [1] '1.1.3'
tar_deps({
  colnames(vacc_vs_unvacc) <- paste0(colnames(vacc_vs_unvacc), ".vacc")
  colnames(tlm_prod_vs_unprod_mixcr) <- paste0(colnames(tlm_prod_vs_unprod_mixcr), ".prod")
  dff <- cbind(vacc_vs_unvacc, tlm_prod_vs_unprod_mixcr) |> 
    as.data.frame()
  dff$gene <- gsub(pattern = "^ENSG[0-9]+_?", "", rownames(dff))
  dff |> 
    ggplot() + 
    geom_point(aes(logFC.vacc, logFC.prod, size = -log10(pmax(FDR.vacc, FDR.prod)))) + 
    geom_text_repel(aes(logFC.vacc, logFC.prod, label = gene), 
                    data = filter(dff, FDR.vacc < 0.1 & FDR.prod < 0.1), col = "red") + 
    theme_minimal() +
    scale_size(range = c(1, 6), breaks = c(1, 2, 4, 6, 8, 10)) +
    labs(x = "logFC Vacc vs unvac", y = "logFC prod vs unprod", 
         title = "Genes expressed in productive and vaccinated cells",
         size = "-log10 FDR")
})
#>  [1] "-"               "{"               "&"               "+"              
#>  [5] "<"               "<-"              "$<-"             "aes"            
#>  [9] "as.data.frame"   "c"               "cbind"           "colnames"       
#> [13] "colnames<-"      "FDR.prod"        "FDR.vacc"        "filter"         
#> [17] "gene"            "geom_point"      "geom_text_repel" "ggplot"         
#> [21] "gsub"            "labs"            "log10"           "logFC.prod"     
#> [25] "logFC.vacc"      "paste0"          "pmax"            "rownames"       
#> [29] "scale_size"      "theme_minimal"
```

Note that `vacc_vs_unvacc` and `tlm_prod_vs_unprod_mixcr` (my data.frames with data) do not appear in the output of the dependencies. This results in an error when attempting to generate this one.

I'm not sure if I should move to use only functions or this should work. I tracked down how this vector is obtained and it comes from codetools. However, codetools has many [bugs reported and not fixed](https://gitlab.com/luke-tierney/codetools/-/issues) and might be related to [this bug](https://gitlab.com/luke-tierney/codetools/-/issues/4).

**Comment by wlandau** [comment]
This is the same issue as https://github.com/ropensci/targets/issues/1088, somehow `codetools` has a hard time handling calls to replacement functions. (The bug you linked seems relevant too.) I recommend the same workaround as the original poster of that thread: https://github.com/nptscot/npt/commit/0645362596c88d71e3d0834f74618efecc0fff0b. Short of switching `targets` from `codetools` to a different code analysis package like `globals`, which would invalidate lots of pipelines, I'm afraid my hands are tied here.

**Reply by llrs** [reply]
Sorry I didn't find the duplicate question. Thanks for the link to the workaround. 
It seems sensible to push for further updates in codetools because it is also used by `R CMD check`, but I understand the feeling. Thanks as always!
